diff -Naur a/Makefile b/Makefile
--- a/Makefile	2017-03-09 16:17:23.000000000 -0800
+++ b/Makefile	2017-05-31 16:04:29.052439734 -0700
@@ -184,7 +184,7 @@
 #
 LIBFDT_dir = libfdt
 LIBFDT_archive = $(LIBFDT_dir)/libfdt.a
-LIBFDT_lib = $(LIBFDT_dir)/libfdt-$(DTC_VERSION).$(SHAREDLIB_EXT)
+LIBFDT_lib = $(LIBFDT_dir)/libfdt.$(SHAREDLIB_EXT)
 LIBFDT_include = $(addprefix $(LIBFDT_dir)/,$(LIBFDT_INCLUDES))
 LIBFDT_version = $(addprefix $(LIBFDT_dir)/,$(LIBFDT_VERSION))
