diff --git a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
index 82bf70596b..ebe57842f6 100644
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
@@ -115,6 +115,7 @@ 
   virtual int codec_set_cntl_mode(codec_para_t *pcodec, unsigned int mode)=0;
   virtual int codec_set_cntl_avthresh(codec_para_t *pcodec, unsigned int avthresh)=0;
   virtual int codec_set_cntl_syncthresh(codec_para_t *pcodec, unsigned int syncthresh)=0;
+  virtual int codec_set_freerun_mode(codec_para_t *pcodec, unsigned int mode)=0;
 
   virtual int codec_set_av_threshold(codec_para_t *pcodec, int threshold)=0;
   virtual int codec_set_video_delay_limited_ms(codec_para_t *pcodec,int delay_ms)=0;
@@ -142,6 +143,7 @@ 
   DEFINE_METHOD2(int, codec_set_cntl_mode,      (codec_para_t *p1, unsigned int p2))
   DEFINE_METHOD2(int, codec_set_cntl_avthresh,  (codec_para_t *p1, unsigned int p2))
   DEFINE_METHOD2(int, codec_set_cntl_syncthresh,(codec_para_t *p1, unsigned int p2))
+  DEFINE_METHOD2(int, codec_set_freerun_mode,   (codec_para_t *p1, unsigned int p2))
 
   DEFINE_METHOD2(int, codec_set_av_threshold,   (codec_para_t *p1, int p2))
   DEFINE_METHOD2(int, codec_set_video_delay_limited_ms, (codec_para_t *p1, int p2))
@@ -164,6 +166,7 @@ 
     RESOLVE_METHOD(codec_set_cntl_mode)
     RESOLVE_METHOD(codec_set_cntl_avthresh)
     RESOLVE_METHOD(codec_set_cntl_syncthresh)
+    RESOLVE_METHOD(codec_set_freerun_mode)
 
     RESOLVE_METHOD(codec_set_av_threshold)
     RESOLVE_METHOD(codec_set_video_delay_limited_ms)
@@ -1785,7 +1788,7 @@ bool CAMLCodec::OpenAmlVideo(const CDVDStreamInfo &hints)
   m_defaultVfmMap = GetVfmMap("default");
   SetVfmMap("default", "decoder ppmgr deinterlace amlvideo amvideo");

-  SysfsUtils::SetInt("/sys/module/amlvideodri/parameters/freerun_mode", 3);
+  m_dll->codec_set_freerun_mode(&am_private->vcodec, 3);
 
   return true;
 }
diff --git a/xbmc/utils/AMLUtils.cpp b/xbmc/utils/AMLUtils.cpp
index fe1dae0a5c..e4db79ef62 100644
--- a/xbmc/utils/AMLUtils.cpp
+++ b/xbmc/utils/AMLUtils.cpp
@@ -101,11 +101,6 @@ bool aml_permissions()
       CLog::Log(LOGERROR, "AML: no rw on /dev/video10");
       permissions_ok = 0;
     }
-    if (!SysfsUtils::HasRW("/sys/module/amlvideodri/parameters/freerun_mode"))
-    {
-      CLog::Log(LOGERROR, "AML: no rw on /sys/module/amlvideodri/parameters/freerun_mode");
-      permissions_ok = 0;
-    }
     if (!SysfsUtils::HasRW("/sys/class/video/freerun_mode"))
     {
       CLog::Log(LOGERROR, "AML: no rw on /sys/class/video/freerun_mode");
--
